docs/content/developers/controllers/client-go.md

```diff
@@ -0,0 +1,25 @@
+# client-go
+
+kcp maintains its own `client-go` implementation at [github.com/kcp-dev/client-go](https://github.com/kcp-dev/client-go) for multi-cluster-aware Kubernetes clients, listers and informers.
+
+## Keys for Objects in Listers/Indexers
+
+When you need to get an object from a kcp-aware lister or an indexer, you can't just pass the object's name to the
+`Get()` function, like you do with a typical controller targeting Kubernetes. Projects using kcp's copy of client-go
+are using a modified key function.
+
+Here are what keys look like for an object `foo` for both cluster-scoped and namespace-scoped varieties:
+
+|Organization|Workspace|Logical Cluster|Namespace|Key|
+|-|-|-|-|-|
+|-|-|root|-|root|foo|
+|-|-|root|default|default/root|foo|
+|root|my-org|root:my-org|-|root:my-org|foo|
+|root|my-org|root:my-org|default|default/root:my-org|foo|
+|my-org|my-workspace|my-org:my-workspace|-|my-org:my-workspace|foo|
+|my-org|my-workspace|my-org:my-workspace|default|default/my-org:my-workspace|foo|
```

Comment:
q: I am not sure I fully understand this part. I thought so far that the kcp-dev/client-go was supposed to be used by directly scoping down to the logicalcluster. So something like this

```Go

myCm, err := clientset.CoreV1().ConfigMaps().Cluster(logicalcluster.NewPath("mylogicalcluster")).Namespace("myns").Get(ctx, "mycm", metav1.GetOptions{})

// or with a dynamic client

res, err := client.Cluster(logicalcluster.NewPath("mylogicalcluster")).Resource(corev1.SchemeGroupVersion.WithResource("configmaps")).Get(ctx, "mycm", metav1.GetOptions{})
```

Can you maybe share a snipped what the specialized key looks like? Or would I also in the aforementioned examples need to replace "mycm" with the specialized key?

Reply:
I think at the very least the org column should disappear, orgs are not their own concept in kcp, right?

Reply:
Just dropped the whole thing. I think it might have been an earlier version of client-go. I don't think this is relevant to people developing kcp controllers in any way.